### ISSUE TITLE
Parse Query ordering with matches query

### ISSUE THREAD

**Issue by kwstasna** [ParsePlatform/parse-server#2615]
- [x] You've met the prerequisites: https://github.com/ParsePlatform/parse-server/wiki/Parse-Server-Guide#prerequisites.

- [x] You're running the latest version of Parse Server: https://github.com/ParsePlatform/parse-server/releases

- [x] You've searched through existing issues: https://github.com/ParsePlatform/Parse-Server/issues?utf8=%E2%9C%93&q=is%3Aissue Chances are that your issue has been reported or resolved before.

- [x] You have filled out every section below. Issues without sufficient information are more likely to be closed.

--

### Issue Description

[DELETE EVERYTHING ABOVE THIS LINE BEFORE SUBMITTING YOUR ISSUE]

Describe your issue in as much detail as possible.

Hello. I have 2 tables in my Parse Server.

First Table

**Venues**

1. Name (strint)
2. Location (Geopoint)

Second Table

**Events**

1. Name (string)
2. idVenue (Pointer to class Venues)

And when i'm trying to take the Events that are near me with Matches Query (iOS) 
I make a query first in Class Venues

```
        let queryEvents = PFQuery(className: "Events")
        let queryVenues = PFQuery(className: "Venues")
        
        
        queryVenues.whereKey("Location", nearGeoPoint: myGeoPoitz, withinKilometers: 15)
        queryEvents.includeKey("idVenue")
        queryEvents.whereKey("idMagazi", matchesQuery: queryVenues)
```    
        

### Steps to reproduce

Please include a detailed list of steps that reproduce the issue. Include curl commands when applicable.

1. [FILL THIS OUT]
2. [FILL THIS OUT]
3. [FILL THIS OUT]

#### Expected Results

To bring the Events on a certain order that they are near me

#### Actual Outcome

It actually brings events on the order that have been inserted on Class

### Environment Setup

- **Server**
  - parse-server version (Be specific! Don't say 'latest'.) : 2.2.18
  - Operating System:     CentOS 6
  - Hardware:             Don't know
  - Localhost or remote server? (AWS, Heroku, Azure, Digital Ocean, etc):  Remote

- **Database**
  - MongoDB version: 3.0.2

### Logs/Trace

You can turn on additional logging by configuring VERBOSE=1 in your environment.

no point on doing this.
We have to find a way to make 2nd (and nth) levels of queries to have a certain order.
Thanks a lot for your time.

**Comment by flovilmart** [comment]
What was the behaviour on parse.com? What should happen if you specify a different order on your main query?

**Comment by kwstasna** [comment]
It was exactly the same also in parse.com
I'm talking about an improvement here. A feature that didn't exist neither in parse.com
Let me rephrase.

If i want to find events near my location
First i have to get my location
then i have to query Venues class to find which Venues are near me
and then with matches query to find the events that have the Object ID of the Venue inside the collumn "venuesID" (which is a pointer to Class Venues)

But this query will NOT bring me the closest events by order (from nearest to other way).
It will bring me in the order of createdAt.

**Comment by flovilmart** [comment]
Alright so that's an improvement/new feature. Feel free to work on it and submit a PR. For now we prioritize more issues that prevent people from correctly running parse-server or transitioning. 

We welcome additional features, like @cherukumilli is doing for securing user accounts.
